core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1291,13 +1303,30 @@ public boolean maybeAcquireFetchLock() {
         if (stateNotActive()) {
             return false;
         }
-        return fetchLock.compareAndSet(false, true);
+        boolean acquired = fetchLock.compareAndSet(false, true);
+        if (acquired) {
+            long currentTime = time.hiResClockMs();
+            fetchLockAcquiredTimeMs = currentTime;
+            timeSinceLastLockAcquisitionMs = timeSinceLastLockAcquisitionMs != 0 ? currentTime - timeSinceLastLockAcquisitionMs : 0;
```

Comment:
Could we please have some defensive code here to guard against time running backwards? Of course, what I mean is perhaps the time gets adjusted and `currentTime < timeSinceLastLockAcquisitionMs`. I don't think we want a negative value.

Reply:
Though this should not happen in practice, but I have added a safe check with comment.